sdk/src/azure/platform/az_posix.c

```diff
@@ -5,18 +5,22 @@
 #include <azure/core/internal/az_config_internal.h>
 #include <azure/core/internal/az_precondition_internal.h>
 
+#include <pthread.h>
+#include <signal.h>
+#include <stdlib.h>
 #include <time.h>
-
 #include <unistd.h>
 
 #include <azure/core/_az_cfg.h>
 
+void _thread_handler(union sigval sv);
+
 AZ_NODISCARD az_result az_platform_clock_msec(int64_t* out_clock_msec)
 {
   _az_PRECONDITION_NOT_NULL(out_clock_msec);
 
   // NOLINTNEXTLINE(bugprone-misplaced-widening-cast)
-  *out_clock_msec = (int64_t)((clock() / CLOCKS_PER_SEC) * _az_TIME_MILLISECONDS_PER_SECOND);
+  *out_clock_msec = (int64_t)((time(NULL)) * _az_TIME_MILLISECONDS_PER_SECOND);
```

Comment:
Note, this resolution is now in seconds

Reply:
+1 please add a 
```C
// TODO #2364 : Implement runtime detection of the POSIX high-res timer if available (fallback to time(NULL) low res if not).
```

This references https://github.com/Azure/azure-sdk-for-c/issues/2364